### PR TITLE
Data Project

### PR DESCRIPTION
**Pull request by MattEqualsCoder**
This basically just pulls out a lot of the configuration, regions, and things like that into the data project so that they can be shared more amongst all of the projects. Nothing has really been changed to really make use of it so far, but it's possible! Wanted to get this out there sooner rather than later to avoid potential conflicts if you work on anything.

### CODE REVIEW

File: src/Randomizer.App/StringColorConverter.cs [hunk 1]
```diff
@@ -1,29 +1,27 @@
...
+using System.Collections.Generic;
 using System.ComponentModel.DataAnnotations;
 using System.Globalization;
 using System.Windows.Data;
 using System.Windows.Media;
 
 namespace Randomizer.App
 {
-    [ValueConversion(typeof(Color), typeof(string))]
+    [ValueConversion(typeof(List<byte>), typeof(string))]
```
Comment (Vivelin) [inline src/Randomizer.App/StringColorConverter.cs:10]: I assume this is because Color is a WPF-specific class?

List<byte> is a little confusing, `byte[]` would be more appropriate. A Color class of our own would be best, but that's probably a little bit too much for how little we actually use colors.

File: src/Randomizer.App/Windows/GenerateRomWindow.xaml [hunk 2]
```diff
@@ -7,20 +7,20 @@
...
         xmlns:vm="clr-namespace:Randomizer.App.ViewModels"
-        xmlns:smz3="clr-namespace:Randomizer.SMZ3;assembly=Randomizer.SMZ3"
+        xmlns:data="clr-namespace:Randomizer.Data;assembly=Randomizer.Data"
         xmlns:shared="clr-namespace:Randomizer.Shared;assembly=Randomizer.Shared"
-        d:DataContext="{d:DesignInstance vm:RandomizerOptions, IsDesignTimeCreatable=True}"
+        xmlns:options="clr-namespace:Randomizer.Data.Options;assembly=Randomizer.Data"
         mc:Ignorable="d"
         Closing="Window_Closing"
-        Title="SMZ3 Cas' Randomizer"
+        Title="Data Cas' Randomizer"
```
Comment (Vivelin) [inline src/Randomizer.App/Windows/GenerateRomWindow.xaml:15]: I think you might've Found and Replaced a little bit too much

File: src/Randomizer.Data/WorldData/Regions/Region.cs [hunk 3]
```diff
@@ -106,23 +108,23 @@ private bool MatchesItemPlacementRule(Item item)
         {
             var rule = Config.ItemPlacementRule;
             if (rule == ItemPlacementRule.Anywhere
-                || (!item.Progression && !item.IsKey && !item.IsKeycard && !item.IsBigKey)
```
Comment (Vivelin) [inline src/Randomizer.Data/WorldData/Regions/Region.cs:109]: Why did you remove the parentheses here?

**Review by Vivelin — APPROVED** [review]

File: src/Randomizer.Data/WorldData/Regions/Region.cs (hunk 3 above)
Comment (MattEqualsCoder) [reply src/Randomizer.Data/WorldData/Regions/Region.cs:109]: Huh. I definitely don't remember doing this, and it doesn't seem like something I would do. I've noticed visual studio has started to recommend removing of unnecessary parenthesis, so I wonder if it did it automatically when I moved files? These are superfluous parenthesis, but they really help readability IMO.

File: src/Randomizer.App/Windows/GenerateRomWindow.xaml (hunk 2 above)
Comment (MattEqualsCoder) [reply src/Randomizer.App/Windows/GenerateRomWindow.xaml:15]: Good catch. I think this was probably a single file find & replace as a global one like that would have been real dumb plus I think we would have seen a lot more. However, just to be safe I'll review over the changes myself again later when I have time to make sure nothing else like this happened though as a bit of due diligence.

File: src/Randomizer.App/StringColorConverter.cs (hunk 1 above)
Comment (MattEqualsCoder) [reply src/Randomizer.App/StringColorConverter.cs]: Yeah, I tried adding in the framework to the data project after the fact since it wasn't added originally since it was just a library project, and... the App project really didn't seem to like that. So this was sort of the path of least resistance.

Switched to an array, though. The change corrupted my settings config, but I think that shouldn't happen for other folks.

File: src/Randomizer.Data/WorldData/Regions/Region.cs (hunk 3 above)
Comment (Vivelin) [reply src/Randomizer.Data/WorldData/Regions/Region.cs]: Either we need to update the .editorconfig or something's broken. Again.

File: src/Randomizer.App/Windows/GenerateRomWindow.xaml (hunk 2 above)
Comment (Vivelin) [reply src/Randomizer.App/Windows/GenerateRomWindow.xaml]: Yeah, I don't think Data Cas’ Randomizer is the new name we wanted to go for, haha

**Review by Vivelin — DISMISSED** [review]

**Comment by MattEqualsCoder** [comment]
Found another location where parenthesis were removed upon copying over. Other than that, I didn't see any other real issues.

**Review by Vivelin — APPROVED** [review]
